plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -870,6 +871,50 @@ public void testMetadataDeleteAfterCommitEnabled()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testIcebergTablesSystemTable()
+            throws Exception
+    {
+        String schemaName = getSession().getSchema().orElseThrow();
+        String firstSchema = "first_schema_" + randomNameSuffix();
+        String secondSchema = "second_schema_" + randomNameSuffix();
+        String firstSchemaLocation = schemaPath().replaceAll(schemaName, firstSchema);
+        String secondSchemaLocation = schemaPath().replaceAll(schemaName, secondSchema);
+        assertQuerySucceeds("CREATE SCHEMA " + firstSchema + " WITH (location = '%s')".formatted(firstSchemaLocation));
+        assertQuerySucceeds("CREATE SCHEMA " + secondSchema + " WITH (location = '%s')".formatted(secondSchemaLocation));
+        QueryRunner queryRunner = getQueryRunner();
+        Session firstSchemaSession = Session.builder(queryRunner.getDefaultSession()).setSchema(firstSchema).build();
+        Session secondSchemaSession = Session.builder(queryRunner.getDefaultSession()).setSchema(secondSchema).build();
+
+        try (TestTable _ = new TestTable(
+                sql -> getQueryRunner().execute(firstSchemaSession, sql),
+                "first_schema_table1_",
+                "(id int)");
+                TestTable _ = new TestTable(
+                        sql -> getQueryRunner().execute(firstSchemaSession, sql),
+                        "first_schema_table2_",
+                        "(id int)");
+                TestTable secondSchemaTable = new TestTable(
+                        sql -> queryRunner.execute(secondSchemaSession, sql),
+                        "second_schema_table_",
+                        "(id int)");
```

Comment:
Can we change these logic so we can test Unity and Snowflake catalogs? e.g. preparing a helper method and directly creating tables on those catalogs without Trino.

Reply:
Done. I did separate the snowflake test change as it is different from normal snowflake tests as it creates schema and tables.